src/ldp/http/SimpleSparqlUpdateBodyParser.ts

```diff
@@ -0,0 +1,47 @@
+import arrayifyStream from 'arrayify-stream';
+import { BodyParser } from './BodyParser';
+import { HttpRequest } from '../../server/HttpRequest';
+import { Readable } from 'stream';
+import { SparqlUpdatePatch } from './SparqlUpdatePatch';
+import { translate } from 'sparqlalgebrajs';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+import { UnsupportedMediaTypeHttpError } from '../../util/errors/UnsupportedMediaTypeHttpError';
+
+/**
+ * {@link BodyParser} that supports `application/sparql-update` content.
+ * Will convert the incoming update string to algebra in a {@link SparqlUpdatePatch}.
+ * Simple since metadata parsing is not yet implemented.
+ */
+export class SimpleSparqlUpdateBodyParser extends BodyParser {
+  public async canHandle(input: HttpRequest): Promise<void> {
+    const contentType = input.headers['content-type'];
+
+    if (!contentType || contentType !== 'application/sparql-update') {
+      throw new UnsupportedMediaTypeHttpError('This parser only supports SPARQL UPDATE data.');
+    }
+  }
+
+  public async handle(input: HttpRequest): Promise<SparqlUpdatePatch> {
+    try {
+      const sparql = (await arrayifyStream(input)).join('\n');
```

Comment:
I suspect this stringification will be needed by many body parsers. We may want to consider abstracting this in `BodyParser`. (not urgent though)

Reply:
I'm confused by the `\n`; that surely must be a mistake? They should just be concatenated?

But yes, isn't there a simple `arrayToString` somewhere? (And if not, let's write it?)

Reply:
Yea no reason for the `\n` to be there. Probably just seemed nicer in my head at the time to have everything on separate lines.

It is `readableToString` that is needed here (and the function for that would be that line, can put it in util).

Reply:
> no reason

Even worse, it would mess up chunked input 😉